src/metrics/metrics.hpp

```diff
@@ -6,12 +6,29 @@
 
 #pragma once
 
+#include <memory>
+#include <string>
+#include <map>
+
 #include "metrics/registry.hpp"
 
 namespace lean::metrics {
-  // the function recommended to use to create a registry of the chosen
-  // implementation
-  std::unique_ptr<Registry> createRegistry();
+  class GaugeHelper;
+
+  /**
+   * @brief Metrics interface that holds all application metrics
+   *
+   * This interface provides access to application-wide metrics collection.
+   */
+  class Metrics {
+   public:
+    virtual ~Metrics() = default;
+
+    /**
+     * @brief Get the underlying registry
+     */
+    virtual Registry& getRegistry() = 0;
+  };
```

Comment:
Components (e.g. `ForkChoiceStore`) should accept interface `Metrics`,
which should provide virtual methods for all metrics,
to allow mocking.
Then `MetricsImpl` and `MetricsMock` would implement/stub/mock these methods.

Please choose desired interface:
```c++
// metrics.time().add(1);
#define METRIC_GAUGE(name, ...) \
  virtual Gauge &name() = 0
```
```c++
// metrics.add_time(1);
#define METRIC_GAUGE(name, ...) \
  virtual void add_##name(double value) = 0; \
  virtual void set_##name(double value) = 0;
```